platform/website/src/components/settings/account/enable-2fa.svelte

```diff
@@ -0,0 +1,259 @@
+<script lang="ts">
+	import Dialog from "$/components/dialog.svelte";
+	import ShieldCheck from "$/components/icons/settings/shield-check.svelte";
+	import Spinner from "$/components/spinner.svelte";
+	import { graphql } from "$/gql";
+	import { downloadBackupCodes } from "$/lib/twoFa";
+	import { user } from "$/store/auth";
+	import { faArrowRight, faDownload } from "@fortawesome/free-solid-svg-icons";
+	import { CombinedError, getContextClient } from "@urql/svelte";
+	import { createEventDispatcher } from "svelte";
+	import Fa from "svelte-fa";
+
+	const dispatch = createEventDispatcher();
+	const client = getContextClient();
+
+	function clickClose() {
+		if (state.step === 3) {
+			next(false);
+		} else {
+			dispatch("close");
+		}
+	}
+
+	function next(fromForm: boolean) {
+		if (state.step === 1) {
+			generateSecret();
+		} else if (state.step === 2) {
+			if (fromForm) {
+				enableTotp();
+			}
```

Comment:
```suggestion
		} else if (state.step === 2 & fromForm) {
			enableTotp();
```

Reply:
There are other if/elses that could be simplified, but I don't want to be too nit.